napalm/pluribus.py

```diff
@@ -0,0 +1,173 @@
+            'uptime'        : string_parsers.convert_uptime_string_seconds(uptime_str),
+            'model'         : unicode(model),
+            'serial_number' : unicode(serial),
+            'interface_list': interfaces
+        }
+
+        return facts
+
+    def get_bgp_neighbors_detail(self, neighbor_address = ''):
+
+        bgp_neighbors = dict()
+
+        return bgp_neighbors
+
+    def cli(self, command = ''):
+
+        return self.device.cli(command)
+
+    def get_arp_table(self, interface = '', host = '', ip = '', mac = ''):
+
+        arp_table = dict()
+
+        return arp_table
+
+    def get_mac_address_table(self, address = '', interface = '', dynamic = False, static = False, vlan = None):
+
+        mac_table = dict()
+
+        mac_show = self.device.execute_show('l2-table-show')
+        lines = mac_show.split('\n')[1:-1]
+
+        for line in lines:
+            mac_details = line.split(';')
+            mac         = mac_details[2].strip()
+            vlan        = int(mac_details[3].strip())
+            ports       = mac_details[8].strip()
+            active      = (mac_details[9].strip == 'active')
+            hostname    = mac_details[10].strip()
+            status      = mac_details[11].strip()
+            migrate     = mac_details[-1].strip()
+            if vlan not in mac_table.keys():
+                mac_table[vlan] = list()
+            mac_table[vlan].append(
+                {
+                    'mac'       : mac,
+                    'interface' : ports,
+                    'active'    : active,
+                    'hostname'  : hostname,
+                    'status'    : status,
+                    'migrate'   : migrate
+                }
+            )
+
+        return mac_table
+
+    def get_ntp_peers(self):
+
+        ntp_peers = dict()
+
+        return ntp_peers
+
+    def get_lldp_neighbors_detail(self, interface = ''):
+
+        lldp_neighbors = dict()
+
+        lldp_show = self.device.execute_show('lldp-show')
+        lines = lldp_show.split('\n')[1:-1]
+
+        for line in lines:
+            neighbor_details    = line.split(';')
+            port                = neighbor_details[1].strip()
+            chassis             = neighbor_details[2].strip()
+            port_id             = neighbor_details[3].strip()
+            port_descr          = neighbor_details[4].strip()
+            system_name         = neighbor_details[6].strip()
+            if port not in lldp_neighbors.keys():
+                lldp_neighbors[port] = list()
+            lldp_neighbors[port] = {
+                'parent_interface'          : None,
+                'remote_port'               : port_id,
+                'remote_port_name'          : port_descr,
+                'remote_system_chassis_id'  : chassis,
+                'remote_system_name'        : system_name
+            }
+
+        return lldp_neighbors
+
+    def show_route(self, destination = ''):
+
+        route = dict()
```

Comment:
Why are you overriding all these methods if they don't do anything? If they are not supported I would rather leave them out so they throw a NotImplementedException.